plugin.video.vtm.go/resources/lib/plugin.py

```diff
@@ -0,0 +1,207 @@
+    Catalog(kodi).show_program_season(program, int(season))
+
+
+@routing.route('/catalog/recommendations')
+def show_recommendations():
+    """ Shows the programs of a specific date in the tv guide """
+    from resources.lib.modules.catalog import Catalog
+    Catalog(kodi).show_recommendations()
+
+
+@routing.route('/catalog/recommendations/<category>')
+def show_recommendations_category(category):
+    """ Show the items in a recommendations category """
+    from resources.lib.modules.catalog import Catalog
+    Catalog(kodi).show_recommendations_category(category)
+
+
+@routing.route('/catalog/mylist')
+def show_mylist():
+    """ Show the items in "My List" """
+    from resources.lib.modules.catalog import Catalog
+    Catalog(kodi).show_mylist()
+
+
+@routing.route('/catalog/mylist/add/<video_type>/<content_id>')
+def mylist_add(video_type, content_id):
+    """ Add an item to "My List" """
+    from resources.lib.modules.catalog import Catalog
+    Catalog(kodi).mylist_add(video_type, content_id)
+
+
+@routing.route('/catalog/mylist/del/<video_type>/<content_id>')
+def mylist_del(video_type, content_id):
+    """ Remove an item from "My List" """
+    from resources.lib.modules.catalog import Catalog
+    Catalog(kodi).mylist_del(video_type, content_id)
+
+
+@routing.route('/catalog/continuewatching')
+def show_continuewatching():
+    """ Show the items in "Continue Watching" """
+    from resources.lib.modules.catalog import Catalog
+    Catalog(kodi).show_continuewatching()
+
+
+@routing.route('/search')
+@routing.route('/search/<query>')
+def show_search(query=None):
+    """ Shows the search dialog """
+    from resources.lib.modules.search import Search
+    Search(kodi).show_search(query)
+
+
+@routing.route('/metadata/update')
+def metadata_update():
+    """ Update the metadata for the listings (called from settings) """
+    from resources.lib.modules.metadata import Metadata
+    Metadata(kodi).update()
+
+
+@routing.route('/metadata/clean')
+def metadata_clean():
+    """ Clear metadata (called from settings) """
+    from resources.lib.modules.metadata import Metadata
+    Metadata(kodi).clean()
+
+
+@routing.route('/play/epg/<channel>/<timestamp>')
+def play_epg_datetime(channel, timestamp):
+    """ Play a program based on the channel and the timestamp when it was aired """
+    from resources.lib.modules.tvguide import TvGuide
+    TvGuide(kodi).play_epg_datetime(channel, timestamp)
+
+
+@routing.route('/play/catalog/<category>/<item>/<channel>')
+def play_or_live(category, item, channel):
+    """ Ask to play the requested item or switch to the live channel """
+    from resources.lib.modules.player import Player
+    Player(kodi).play_or_live(category, item, channel)
+
+
+@routing.route('/play/catalog/<category>/<item>')
+def play(category, item):
+    """ Play the requested item """
+    from resources.lib.modules.player import Player
+    Player(kodi).play(category, item)
```

Comment:
same story. You can still use the player class to wait for playback to start and feed it subtitiles afterwards but calling play directly has unwanted consequences. Furthermore, you loose features of the plugin addon type such as watch states, resume, etc.

Reply:
As the other remark, this Player().play() is from `resources.lib.modules.player`, and this is just a class to do metadata lookup and in the end uses `setResolvedUrl`.

Reply:
@enen92 Same here.